fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/MetricUnionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,357 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.table.Column
+import com.yahoo.bard.webservice.table.PhysicalTable
+import com.yahoo.bard.webservice.table.TestAvailability
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.Interval
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.util.stream.Collectors
+
+/**
+ * Test for metric union availability behavior.
+ */
+class MetricUnionAvailabilitySpec extends Specification {
+
+    MetricUnionAvailability metricUnionAvailability
+
+    @Shared PhysicalTable table1
+    @Shared PhysicalTable table2
+
+    @Shared Availability availability1
+    @Shared Availability availability2
+
+    Set<Column> columns
+    Column column1
+    Column column2
+
+    DataSourceConstraint dataSourceConstraint
+
+    class TestMetricUnionAvailability extends MetricUnionAvailability {
+
+        List<TestAvailability> testAvailabilities
```

Comment:
Since `Availability` is an interface. An implementation of it, `TestAvailability`, can return specified value from `getAvailableIntervals`. We need `Availability.getAvailableIntervals` to return something in test